central/detection/service/service_impl_test.go

```diff
@@ -711,3 +711,29 @@ func TestGetAppliedNetpolsForDeployment(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Len(t, actual.Policies, 1)
 }
+
+func TestGetPolicyNamesAsSlice(t *testing.T) {
+	cases := map[string]struct {
+		policies map[string]*storage.NetworkPolicy
+		expected []string
+	}{
+		"No policies": {
+			policies: make(map[string]*storage.NetworkPolicy),
+			expected: nil,
+		},
+		"Mutliple policies": {
+			policies: map[string]*storage.NetworkPolicy{"1": {Name: "1"}, "2": {Name: "2"}},
+			expected: []string{"1", "2"},
+		},
```

Comment:
This may flake, cause the map does not preserve the order, while slice does. I think, you need to change the assertion to ignore the order, or sort the slice before asserting.